x-pack/plugins/security_solution/public/overview/components/detection_response/recent_cases_table/use_recent_cases_items.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { useState, useEffect } from 'react';
+
+import { CaseStatuses } from '@kbn/cases-plugin/common';
+import { AllCases } from '@kbn/cases-plugin/common/ui';
+
+import { APP_ID } from '../../../../../common/constants';
+import { useGlobalTime } from '../../../../common/containers/use_global_time';
+import { useKibana } from '../../../../common/lib/kibana';
+import { addError } from '../../../../common/store/app/actions';
+
+import * as i18n from '../translations';
+
+export interface RecentCaseItem {
+  name: string;
+  note: string;
+  createdAt: string;
+  createdBy: string;
+  status: CaseStatuses;
+  id: string;
+}
+
+export interface UseRecentlyCreatedCasesProps {
+  skip: boolean;
+}
+
+export type UseRecentlyCreatedCases = (props: UseRecentlyCreatedCasesProps) => {
+  items: RecentCaseItem[];
+  isLoading: boolean;
+  updatedAt: number;
+};
+
+export const useRecentlyCreatedCases: UseRecentlyCreatedCases = ({ skip }) => {
+  const {
+    services: { cases },
+  } = useKibana();
+  const { to, from } = useGlobalTime();
+  const [isLoading, setIsLoading] = useState(true);
+  const [updatedAt, setUpdatedAt] = useState(Date.now());
+  const [items, setItems] = useState<RecentCaseItem[]>([]);
+
+  // This is a known issue of cases api, it doesn't accept date time format atm
+  // Once they fix this problem we can remove this two lines
+  // [TO-DO] #130979
+  const fromDate = moment(from).format('YYYY-MM-DD');
+  const toDate = moment(to).format('YYYY-MM-DD');
+
+  useEffect(() => {
+    let isSubscribed = true;
+    const abortCtrl = new AbortController();
+
+    const fetchCases = async () => {
+      try {
+        const casesResponse = await cases.api.cases.find({
+          from: fromDate,
+          to: toDate,
+          owner: APP_ID,
+          sortField: 'create_at',
+          sortOrder: 'desc',
+          page: 1,
+          perPage: 4,
+        });
+
+        if (isSubscribed) {
+          setItems(parseRecentCases(casesResponse as unknown as AllCases));
```

Comment:
The type of `casesResponse` seemed to be different from `AllCases`

```
export interface CasesStatus {
  countClosedCases: number | null;
  countOpenCases: number | null;
  countInProgressCases: number | null;
}

export interface AllCases extends CasesStatus {
  cases: Case[];
  page: number;
  perPage: number;
  total: number;
}
```

I logged the `casesResponse`, and it was
```
{
  "page": 1,
  "per_page": 4,
  "total": 3,
  "cases": [
    {
      "id": "c1f50550-c562-11ec-95ec-bb25c9b954e6",
      "version": "WzY4ODIsMV0=",
      "comments": [],
      "totalComment": 0,
      "totalAlerts": 0,
      "title": "case 3",
      "tags": [
        "case"
      ],
      "description": "case",
      "settings": {
        "syncAlerts": true
      },
      "owner": "securitySolution",
      "closed_at": null,
      "closed_by": null,
      "created_at": "2022-04-26T13:14:09.320Z",
      "created_by": {
        "username": "elastic",
        "email": null,
        "full_name": null
      },
      "status": "open",
      "updated_at": null,
      "updated_by": null,
      "connector": {
        "id": "none",
        "name": "none",
        "type": ".none",
        "fields": null
      },
      "external_service": null
    },
    {
      "id": "b9cfe840-c562-11ec-95ec-bb25c9b954e6",
      "version": "WzY4ODUsMV0=",
      "comments": [],
      "totalComment": 0,
      "totalAlerts": 0,
      "title": "case 2",
      "tags": [
        "case"
      ],
      "description": "case",
      "settings": {
        "syncAlerts": true
      },
      "owner": "securitySolution",
      "closed_at": null,
      "closed_by": null,
      "created_at": "2022-04-26T13:13:55.654Z",
      "created_by": {
        "username": "elastic",
        "email": null,
        "full_name": null
      },
      "status": "in-progress",
      "updated_at": "2022-04-26T13:14:19.381Z",
      "updated_by": {
        "full_name": null,
        "email": null,
        "username": "elastic"
      },
      "connector": {
        "id": "none",
        "name": "none",
        "type": ".none",
        "fields": null
      },
      "external_service": null
    },
    {
      "id": "1aeed790-c562-11ec-95ec-bb25c9b954e6",
      "version": "WzY4NzksMV0=",
      "comments": [],
      "totalComment": 0,
      "totalAlerts": 0,
      "title": "case",
      "tags": [
        "case"
      ],
      "description": "case",
      "settings": {
        "syncAlerts": true
      },
      "owner": "securitySolution",
      "closed_at": null,
      "closed_by": null,
      "created_at": "2022-04-26T13:09:29.103Z",
      "created_by": {
        "username": "elastic",
        "email": null,
        "full_name": null
      },
      "status": "open",
      "updated_at": null,
      "updated_by": null,
      "connector": {
        "id": "none",
        "name": "none",
        "type": ".none",
        "fields": null
      },
      "external_service": null
    }
  ],
  "count_open_cases": 2,
  "count_in_progress_cases": 1,
  "count_closed_cases": 0
}
```